packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_GenericDistribution.res

```diff
@@ -66,6 +66,38 @@ module Helpers = {
       dist1,
     )->runGenericOperation
   }
+  let parseNumber = (args: expressionValue) : Belt.Result.t<float, string> => 
+    switch args {
+    | EvNumber(x) => Ok(x)
+    | _ => Error("Not a number")
+    }
+
+  let parseNumberArray = (ags: array<expressionValue>) : Belt.Result.t<array<float>, string> => E.A.fmap(parseNumber, ags) |> E.A.R.firstErrorOrOpen
+
+  let parseDist = (args: expressionValue): Belt.Result.t<GenericDist_Types.genericDist, string> => 
+    switch args {
+    | EvDistribution(x) => Ok(x)
+    | EvNumber(x) => Ok(GenericDist.fromFloat(x))
+    | _ => Error("Not a distribution")
+    }
+
+  let parseDistributionArray = (ags: array<expressionValue>) : Belt.Result.t<array<GenericDist_Types.genericDist>, string> => E.A.fmap(parseDist, ags) |> E.A.R.firstErrorOrOpen
+
+  let mixture = (args : array<expressionValue>): DistributionOperation.outputType => {
+    let givenWeights = E.A.last(args)
+    let calculatedWeights = 
+        switch givenWeights {
+        | Some(EvArray(b)) => parseNumberArray(b)
+        | None =>
+            Ok(Belt.Array.make(E.A.length(args), 1.0 /. Belt.Int.toFloat(E.A.length(args))))
+        | _ => Error("Last argument of mx must be array")
+        }
+    switch (parseDistributionArray(Belt.Array.slice(args, ~offset=0, ~len=Belt.Array.length(args)-1)), calculatedWeights) {
```

Comment:
My quick read is that if a final array *isn't* passed in, like,
``
mx(normal(5,2), normal(10,1))
``
,
Then this will either error or ignore the last distribution.

The intended functionality is to just weight them the same.

(I realize now that this use case hasn't been tested yet)

Reply:
Yes! This is true. I'm aware of this but forgot that I need to fix this before merging it in

Reply:
behavior-wise I like unweighted average in this use case.

Reply:
I fixed this in the recent commit!